doc/API.md

```diff
@@ -309,6 +309,15 @@ The response is the same as for [`createspend`](#createspend).
 
 ### `startrescan`
 
+Start a rescan of the block chain for transactions involving our wallet's main descriptor between
+the given date and the current tip.
+
+This command will error if any of the following conditions hold:
+* there is already a rescan in progress
+* provided timestamp is in the future
+* provided timestamp is before the timestamp of the genesis block or, if using a pruned node, the earliest block available
+* if using Electrum as a backend (since a full scan is performed at wallet creation and so rescans are not required)
```

Comment:
I don't think "this is unnecessary" is a good reason to make it return an error. Plus it's useful to debug (for instance in support), and it requires disabling a number of functional tests. Why can't the rescan command just trigger a full rescan?

Reply:
We can add this in a follow-up. I was considering something like a full scan, but it didn't seem completely straightforward to integrate into the wallet logic so I decided to wait in case it wasn't needed.